### ISSUE TITLE
Establish a proposal process for new components

### ISSUE THREAD

**Issue by bryanrsmith** [Faithlife/styled-ui#138]
I think it would be beneficial to document a process for new component contributions to this project. Something like:

* Specifically describe the problem we'd like the new component to solve.
* Investigate existing solutions, and establishing the goals and limitations of our solution.
* Describing the API.
* Invite feedback.
* Implement.

This could help ensure new components adequately cover our use cases and have considered more of the issues that may lead to breaking changes in the future.

**Comment by awesomebob** [comment]
Sounds like a good idea. This would expand upon step 1 under the "How to Contribute" section.
https://github.com/Faithlife/styled-ui#how-to-contribute

**Comment by korbinancell** [comment]
Perhaps Joylent's RFD format https://git/Logos/PayFacApi/tree/master/rfd from https://github.com/joyent/rfd

**Comment by bryanrsmith** [comment]
Perhaps we should start with something informal, and agree to hold a discussion in a new issue before starting development of new components?

**Comment by korbinancell** [comment]
Seems reasonable. Would this live as a pr or an issue? Or as a flmessages thread?

**Comment by bryanrsmith** [comment]
I think a github issue would work well. A message thread is too exclusive, and discussions aren't as easy to track in PRs.

**Comment by korbinancell** [comment]
@bryanrsmith does the new RFD format cover what you wanted or now that it has been used a bit is there any changes you would want to see to it? cc @RobertBolender

**Comment by bryanrsmith** [comment]
I think it's been very helpful! Two things I think have been a little thin, mostly because I was super vague when I wrote them:

1. Investigate existing solutions -- I think we should be looking outside our organization at open source projects before making a proposal. We should consider whether or not an existing solution is a good fit that we could build on; learn from their API decisions; consider what problems we need to solve that they don't, and vice versa; consider what we might need to do differently.
2. Describing the API in detail. Personally, I have a hard time thinking through use cases when I can only see prop types, or vague lists of components. If I can see an actual code sample that would appear in a real application, it's much easier for me to think about how needs might change over time, or how various applications have different requirements. I always write consuming code before I start working on a component's implementation because it's easier for me to play with the API and see what feels right. I'm curious if others get as much value as I do from this, but I'd vote for including a _complete_ code sample that fully uses the proposed component's API.

**Comment by korbinancell** [comment]
I added some of your thoughts as comments with guidelines to the RFD template. Let me know what you think.

**Comment by RobertBolender** [comment]
Without objection, closing this issue as we now have an established RFD issue template. If we need to discuss it further, we can always re-open the issue.
